docs/BUILD.md

```diff
@@ -0,0 +1,163 @@
+
+Build Clarity Angular by running `npm run angular:build`, which calls the following tasks to build the package.
+
+* `angular:build` - Build the Angular package for production
+* `angular:start` - Run the test project for Clarity Angular Components
+* `angular:test` - Run all tests for Clarity Angular
+
+## `@clr/core`
+
+Build Clarity Core by running `npm run core:build`, which calls the following tasks to build the package.
+
+* `core:build` - Builds the Core package for production
+* `core:start` - Run the test project for Clarity Core Web Components
+* `core:test` - Run all tests for Clarity Core
+
+## `@clr/icons`
+
+Build Clarity Icons by running `npm run icons:build`, which calls the following tasks to build the package.
+
+* `icons:build:web` - Webpack compiles and bundles the TypeScript assets
+* `icons:build:css` - Sass compiles the styles
+* `icons:build:optimize` - CSSO optimizes the CSS
+* `icons:build:package` - Copy the `package.json` into the package, and set the version number
+* `icons:build:web` - Build the raw svg files and zip directories for designers
+
+## `@clr/ui`
+
+Build Clarity UI is by running `npm run ui:build`, which calls the following tasks to build the package.
+
+* `ui:build:css` - Sass compiles the light and dark theme files
+* `ui:build:prefix` - Autoprefixer adds prefixes to CSS properties based on browser compatibility
+* `ui:build:src` - Copy in the source files for anyone building directly
+* `ui:build:optimize` - CSSO Optimize the CSS
+* `ui:build:package` - Copy the `package.json` into the package, and set the version number
+
+## Globally Installed NPM packages
+
+The following packages are installed globally in the development environment. The purpose of each is listed below.
+You won't need to install these for general development but may wish to do so if you want to run specific scripts for testing or publishing that require them:
+
+* [@angular/cli](https://cli.angular.io/): The whole project uses this for build, preview, and testing.
+* [gemini](https://gemini-testing.github.io/): this is used to run cli commands to run visual diff regression tests.
+* [html-reporter](https://www.npmjs.com/package/html-reporter): plugin for Gemini to produce an HTML report of the CSS regression tests.
+* [surge](https://surge.sh/): for publishing static website. We use this to publish the kitchen sink app.
+
+## Additional NPM Scripts
+
+There are a few other NPM scripts that can be useful during build and development.
+
+##### `npm start`
+
+The start command starts up our demo app using the Angular CLI on port 4200 and watches for file changes for live reload.
+
+##### `npm run build`
+
+This script builds npm package candidates for all three packages we currently publish: `@clr/angular`, `@clr/ui`, and
+`@clr/icons` under the `/dist` folder.
+
+##### `npm test` and `npm run angular:test:watch`
+
+The `test` script runs the unit tests using karma. The entry point for the tests is `tests/tests.entry.ts`.
```

Comment:
This entry file now lives in `src/clr-angular/test.ts`and is now confined to what's under `clr-angular`. It used to be that the file was a single entry point across multiple packages.

Also, do we want to keep the section specifically about `angular:test:watch`? There's also now `core:test:watch` so we can either mention both or remove it from this section.